### ISSUE TITLE
Angular 12 compiling error on development mode 'Error: Type AbilityModule does not have 'ɵmod' property'

### ISSUE THREAD

**Issue by detonimarco** [stalniy/casl#512]
**Describe the bug**
If I run project in development mode using "ng serve", project doesn't start raising 
```typescript
 core.js:1119 Uncaught Error: Type AbilityModule does not have 'ɵmod' property.
    at getNgModuleDef (core.js:1119)
    at recurse (core.js:25261)
    at recurse (core.js:25272)
    at registerNgModuleType (core.js:25257)
    at new NgModuleFactory$1 (core.js:25371)
    at compileNgModuleFactory__POST_R3__ (core.js:29063)
    at PlatformRef.bootstrapModule (core.js:29309)
    at Module.4431 (main.ts:11)
    at __webpack_require__ (bootstrap:19)
    at __webpack_exec__ (main.js:526)
VM10:87 Uncaught TypeError: Cannot read property 'postMessage' of null
    at notifyFrames (eval at <anonymous> (content.js:7), <anonymous>:87:45)
```

If you run the project settings, inside angular.json the properties:
```json
"development": {
 "buildOptimizer": true,
 "optimization": true,
...
```
as in production configuration it works

**To Reproduce**
1. create a plain Angular 12 project using  "ng new my-project"
2. add casl to project "npm install @casl/angular @casl/ability"
3. configure AppModule as per example: 
```typescript
import { NgModule } from '@angular/core';
import { AbilityModule } from '@casl/angular';
import { Ability, PureAbility } from '@casl/ability';

@NgModule({
  imports: [
    // other modules
    AbilityModule
  ],
  providers: [
    { provide: Ability, useValue: new Ability() },
    { provide: PureAbility, useExisting: Ability }
  ]
  // other properties
})
export class AppModule {}
```
4. run with "ng serve"

**Expected behavior**
The application should start in development mode without exception

**CASL Version**

`@casl/angular` - v

**Environment**:

package.json (already attached)
```json
....
"@angular/core": "~12.0.1",
"@angular/compiler": "~12.0.1",
....

"@casl/ability": "^5.3.1",
"@casl/angular": "^5.1.1",
```
Angular 12.0.1
Node v12.18.3

[package.json.txt](https://github.com/stalniy/casl/files/6531557/package.json.txt)

**Comment by stalniy** [comment]
Hi thanks for the issue.

Angular 12 is not yet supported. casl/angular is built using ViewEngine, not Ivy

**Comment by claudiobianco** [comment]
I've added

`@Injectable()`
before `class AbilityPipe` and it seems to work.

Not sure if that is the right/whole fix though.

**Comment by claudiobianco** [comment]
I created two PRs. Didn't know if angular update and fix should be done separately. 

#513 
#514 

As mentioned in the PR I wasn't sure how to build the package and import it into a project. So I tested it with a local copy of pipes.ts.

**Comment by detonimarco** [comment]
thanks @claudiobianco I'm trying to build locally this days making some tests

**Comment by stalniy** [comment]
I’ll check this on weekend

**Comment by sourav-bhar** [comment]
> I’ll check this on weekend

Thanks in advance! I'm hoping to use CASL for a new Angular 12 project and really looking forward to having this work with Angular 12.

**Comment by stalniy** [comment]
fixed in @casl/agnular@6.0.0 - see release - https://github.com/stalniy/casl/releases/tag/%40casl%2Fangular%406.0.0

**Comment by stalniy** [comment]
I've also upgraded angular-todo example to v12 and @casl/angular@6.0.0
